sql/src/main/java/io/crate/analyze/SetStatementAnalyzer.java

```diff
@@ -78,6 +93,41 @@ public static SetAnalyzedStatement analyze(SetStatement node) {
         }
     }
 
+    public SetLicenseAnalyzedStatement analyze(SetLicenseStatement node, Analysis context) {
+        if (node.assignments().size() != SetLicenseAnalyzedStatement.LICENSE_TOKEN_NUM) {
+            throw new IllegalArgumentException(String.format(Locale.ENGLISH,
+                "Invalid number of settings for SET LICENSE. Please provide %d settings.",
+                SetLicenseAnalyzedStatement.LICENSE_TOKEN_NUM));
+        }
+
+        final FieldProvider fieldProvider = new FieldProvider() {
```

Comment:
I guess we could have a `FieldProvider.UNSUPPORTED` static default implementation on the FieldProvider. We've this ad-hoc defnition elsewhere as well.